lib/chef/provisioning/convergence_strategy/precreate_chef_objects.rb

```diff
@@ -226,10 +238,43 @@ def client_rb_content(chef_server_url, node_name)
             https_proxy #{convergence_options[:bootstrap_proxy].inspect}
           EOM
         end
+        if convergence_options[:policy_name]
+          content << <<-EOM
+            # Policyfile Settings:
+            use_policyfile true
+            policy_document_native_api true
+            policy_group "#{convergence_options[:policy_group]}"
+            policy_name "#{convergence_options[:policy_name]}"
+          EOM
+        end
+
         content.gsub!(/^\s+/, "")
         content << convergence_options[:chef_config] if convergence_options[:chef_config]
         content
       end
+
+      def validate_policy_if_enabled
+        raise "policy_name was given but no policy_group. Both are required" if (convergence_options[:policy_name] &&
+                                                                                 !convergence_options[:policy_group])
+        raise "policy_group was given but no policy_name. Both are required" if (convergence_options[:policy_group] &&
+                                                                                 !convergence_options[:policy_name])
+        if convergence_options[:policy_name] && convergence_options[:chef_config]
```

Comment:
This `if` check isn't necessary since you will have already raised an exception if this isn't true